frontend/src/js/external-forms/helper.ts

```diff
@@ -1,10 +1,27 @@
 import type { SelectOptionT } from "../api/types";
 import type { Language } from "../localization/useActiveLang";
 
-import type { FormField, GeneralField, Group } from "./config-types";
+import type {
+  Field,
+  FormField,
+  GeneralField,
+  Group,
+  Tabs,
+} from "./config-types";
 
 const nonFormFieldTypes = new Set(["HEADLINE", "DESCRIPTION"]);
 
+// Different forms may have fields with the same name.
+// We want to remember values of fields of form A when switching to form B,
+// so users may come back to form A and see their previous values.
+// So in order to avoid field name clashes, we need unique field names
+export const getUniqueFieldname = (formType: string, rawFieldName: string) => {
```

Comment:
Might be a nitpick, but I think might be useful: can you make `getUniqueFieldname` return `uniqueFieldNameT` so that the string-type is tagged on the type system level or would that require you down-casting it when interacting with the form library/react? That way you're less likely to miss casts or misshandle the values. so `getUniqueFieldname : (string, string) -> uniqueFieldnameT` and `getRawFieldname: uniqueFieldnameT -> string`

and `type uniqueFieldnameT = string`

Reply:
I did way too much googling out of curiosity, and there seems to be an approach here: https://www.typescriptlang.org/play#example/nominal-typing

It makes sense though, that this isn't a primary concern of TS (i honestly forgot it was structurally typed)